Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -774,8 +784,7 @@ private class MutableArray: BSONValue {
 private class MutableDictionary: BSONValue {
     fileprivate static var bsonType: BSONType { .document }
 
-    // This is unused
-    fileprivate var bson: BSON { .document(self.toDocument()) }
+    fileprivate var bson: BSON { .document((try? self.toDocument()) ?? [:]) }
```

Comment:
This is used in testing, not sure what to do here since prop getters can't throw, do y'all think this is fine? maybe I can add a comment?